crypto.go

```diff
@@ -616,3 +618,50 @@ func makeTrafficKeys(params CipherSuiteParams, secret []byte) keySet {
 		iv:     HkdfExpandLabel(params.Hash, secret, "iv", []byte{}, params.IvLen),
 	}
 }
+
+func MakeNewSelfSignedCert(name string, alg SignatureScheme) (crypto.Signer, *x509.Certificate, error) {
```

Comment:
I don't think generating a self-signed certificate should be part of a TLS library.

Reply:
Yeah, I obviously don't agree with this. We have two test programs that rely on it and I also use it in Minq. So, I really don't think it's sensible to inconvenience people out of a sense of API purity.

Reply:
+1 to EKR.  We've already crossed the bridge of importing PKI gunk by following the `crypto/tls` approach of having a built-in validation system.  This is a sensible utility to offer folks.